File: src/main/java/com/attilapalf/exceptional/controllers/ExceptionController.java
package com.attilapalf.exceptional.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.attilapalf.exceptional.messages.BaseExceptionRequest;
import com.attilapalf.exceptional.messages.ExceptionInstanceWrapper;
import com.attilapalf.exceptional.messages.ExceptionRefreshResponse;
import com.attilapalf.exceptional.messages.ExceptionSentResponse;
import com.attilapalf.exceptional.services.ExceptionService;
import com.attilapalfi.exceptional.model.Question;

/**
 * Created by Attila on 2015-06-13.
 */
@RestController
public class ExceptionController {

    @Autowired
    private ExceptionService exceptionService;

    @RequestMapping( value = "/exception", method = RequestMethod.POST )
    public ResponseEntity<ExceptionSentResponse> throwException( @RequestBody ExceptionInstanceWrapper exceptionInstanceWrapper ) {
        Question question = exceptionInstanceWrapper.getQuestion();
        if ( question.getText().length() > 150 ) {
            return new ResponseEntity<>( HttpStatus.BAD_REQUEST );
        }
        return new ResponseEntity<>( exceptionService.throwException( exceptionInstanceWrapper ), HttpStatus.OK );
    }

    @RequestMapping( value = "/exception/refresh", method = RequestMethod.POST )
    public ResponseEntity<ExceptionRefreshResponse> refreshExceptions( @RequestBody BaseExceptionRequest requestBody ) {
        return new ResponseEntity<>( exceptionService.refreshExceptions( requestBody ), HttpStatus.OK );
    }
}

File: src/main/java/com/attilapalf/exceptional/controllers/ExceptionController.kt
package com.attilapalf.exceptional.controllers

import com.attilapalf.exceptional.messages.*
import com.attilapalf.exceptional.services.ExceptionService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

/**
 * Created by Attila on 2015-06-13.
 */
@RestController
public class ExceptionController {

    @Autowired
    private lateinit val exceptionService: ExceptionService

    @RequestMapping(value = "/exception", method = arrayOf(RequestMethod.POST))
    fun throwException(@RequestBody wrapper: ExceptionInstanceWrapper): ResponseEntity<ExceptionSentResponse> {
        val question = wrapper.question
        if (question.text != null && question.text.length() > 100) {
            return ResponseEntity(HttpStatus.BAD_REQUEST)
        }
        return ResponseEntity(exceptionService.throwException(wrapper), HttpStatus.OK)
    }

    @RequestMapping(value = "/exception/refresh", method = arrayOf(RequestMethod.POST))
    fun refreshExceptions(@RequestBody requestBody: BaseExceptionRequest): ResponseEntity<ExceptionRefreshResponse> {
        return ResponseEntity(exceptionService.refreshExceptions(requestBody), HttpStatus.OK)
    }

    @RequestMapping(value = "/exception/answer", method = arrayOf(RequestMethod.POST))
    fun answerQuestion(@RequestBody questionAnswer: QuestionAnswer): ResponseEntity<ExceptionSentResponse> {
        return ResponseEntity(exceptionService.answerQuestion(questionAnswer), HttpStatus.OK)
    }
}
